### ISSUE TITLE
repo2docker issues

### ISSUE THREAD

**Issue by jameshowison** [softcite/mentions_pipeline_notebook#1]
I played around a bit with the mentions_pipeline_notebook repo.  The “requirements.txt” error was coming from the setup.py specifying software_mentions_client==0.1.0.  If that changes to software_mentions_client==0.1.6 then we get past that bit.  

That said, it then runs into an error that i think happens because python 3.7 is default (for repo2docker, at least with the version on pypy currently.  This causes the require_python='>=3.8' to not pass, stopping the build.

I poked around a bit: https://discourse.jupyter.org/t/can-repo2docker-resolve-python-requires-in-setup-py/18599 and it seems unlikely that repo2docker will ever read the python version out of setup.py (or other python packaging specifications).

Looks like the way forward would be to specify python > 3.8 in an environment.yml file (so that conda handles that install).  

We might move the python dependencies into the requirements.txt file.  I'm wondering whether a module for this notebook makes sense anyway?  Can't see that anyone would install it as a module?

**Comment by kermitt2** [comment]
> We might move the python dependencies into the requirements.txt file. I'm wondering whether a module for this notebook makes sense anyway? Can't see that anyone would install it as a module?

Yes you're right ! I have created a setup.py more as a routine I think but this is not relevant here. We can just have a requirements.txt. I think the default python 3.7 is working well for these modules, so we can also just forget about a minimum requirement for python (I think the python version can be set in a file `runtime.txt` coming with the `requirements.txt`, or everything in `environment.yml` with conda)
